tests/unit/MDFTextAccessibilityUnitTests.swift

```diff
@@ -170,4 +170,46 @@ class MDFTextAccessibilityUnitTests: XCTestCase {
     XCTAssertFalse(MDFTextAccessibility.isLargeForContrastRatios(nil))
   }
 
+  // MARK: textColorOnBackgroundImage
+  
+  func testTextColorOnEmptyBackgroundImage() {
+    let color = MDFTextAccessibility.textColorOnBackgroundImage(UIImage.init(), inRegion:CGRectMake(0, 0, 10, 10), targetTextAlpha: 1.0, font: UIFont.boldSystemFontOfSize(13))
+    XCTAssertNil(color)
+  }
+  
+  func testTextColorOnNonEmptyBackgroundImageOffRegion() {
+    let image = UIImage.init(named: "100_100_gray", inBundle: NSBundle.init(forClass: self.dynamicType), compatibleWithTraitCollection: nil)!
+    let color = MDFTextAccessibility.textColorOnBackgroundImage(image, inRegion:CGRectMake(-10, -10, 10, 10), targetTextAlpha: 1.0, font: UIFont.boldSystemFontOfSize(13))
+    XCTAssertNil(color)
+  }
+  
+  func testTextColorOnNonEmptyBackgroundImageZeroRect() {
+    let image = UIImage.init(named: "100_100_gray", inBundle: NSBundle.init(forClass: self.dynamicType), compatibleWithTraitCollection: nil)!
+    let color = MDFTextAccessibility.textColorOnBackgroundImage(image, inRegion:CGRectZero, targetTextAlpha: 1.0, font: UIFont.boldSystemFontOfSize(13))
+    XCTAssertNil(color)
+  }
+  
+  func testTextColorOnNonEmptyBackgroundImage() {
+    let image = UIImage.init(named: "100_100_gray", inBundle: NSBundle.init(forClass: self.dynamicType), compatibleWithTraitCollection: nil)!
+    let color = MDFTextAccessibility.textColorOnBackgroundImage(image, inRegion:CGRectMake(0, 0, 10, 10), targetTextAlpha: 1.0, font: UIFont.boldSystemFontOfSize(13))
+    XCTAssertNotNil(color)
+  }
+  
+  func testWhiteBackgroundImage() {
+    let image = UIImage.init(named: "100_100_white", inBundle: NSBundle.init(forClass: self.dynamicType), compatibleWithTraitCollection: nil)!
+    let color = MDFTextAccessibility.textColorOnBackgroundImage(image, inRegion:CGRectMake(0, 0, 10, 10), targetTextAlpha: 1, font: UIFont.boldSystemFontOfSize(13))
+    let components = CGColorGetComponents(color?.CGColor)
+    // 0 here for the first element in the components array represents a black color to give the most contrast against a fully white background image
+    XCTAssertTrue(components[0] == 0)
+    XCTAssertTrue(components[1] == 1)
+  }
+  
+  func testBlackBackgroundImage() {
+    let image = UIImage.init(named: "100_100_black", inBundle: NSBundle.init(forClass: self.dynamicType), compatibleWithTraitCollection: nil)!
+    let color = MDFTextAccessibility.textColorOnBackgroundImage(image, inRegion:CGRectMake(0, 0, 10, 10), targetTextAlpha: 1, font: UIFont.boldSystemFontOfSize(13))
+    let components = CGColorGetComponents(color?.CGColor)
+    // 1 here for the first element in the components array represents a white color to give the most contrast against a fully black background image
+    XCTAssertTrue(components[0] == 1)
+    XCTAssertTrue(components[1] == 1)
+  }
```

Comment:
These last two tests may be slightly more clear if you store the target text alpha in a variable and reuse it for the argument and the assertion.